src/odemis/driver/xt_client.py

```diff
@@ -2268,19 +2337,20 @@ def __init__(self, name, role, parent, channel_name, address, **kwargs):
             raise HwError("Failed to connect to XT server '%s'. Check that the "
                           "uri is correct and XT server is"
                           " connected to the network. %s" % (address, err))
-        Detector.__init__(self, name, role, parent=parent, channel_name=channel_name, **kwargs)
+        Detector.__init__(self, name, role, parent=parent, **kwargs)
         self._shape = (2**16,)  # Depth of the image
+        self._scanner = self.parent._scanner
```

Comment:
The parent class already takes care of setting this variable. So you shouldn't change it. => drop thsi line.